### DISCUSSION TITLE
Collate some model checkers for future C++ evaluation

### DISCUSSION THREAD

**Discussion by kunjsong01** [esbmc/esbmc#1225] [Ideas]
Just a general discussion to jot down the idea so that we could have a non-issue ticket to track this task: 

Our previous STVR'21 paper compared ESBMC++ to LLBMC, DIVINE and CBMC. CBMC is still being maintained, which is "downloadable", "buildable" and still functional.

As for LLBMC and DIVINE...

### LLBMC: 
Latest release dated back in 2013, and now not really downloadable:
![Screenshot 2023-07-26 at 15 10 55](https://github.com/esbmc/esbmc/assets/32592800/901d599e-a0ac-459f-9dfd-e466f43f967f)

### DIVINE:
The last commit in the mirror repository on Github was about 2.5 years ago back in March 2021. The download links in https://divine.fi.muni.cz/get.html also stopped working - nothing happened after clicking them. 

Had to download the latest release (Feb 2020, but now July 2023) from https://github.com/paradise-fi/divine/tags,
~~and build from source (which is also a pretty lengthy process).~~ Edit: following the build instructions in https://github.com/esbmc/esbmc/discussions/1225#discussion-5442985, the last time I tried to build it failed with the error: 
```
-- Performing Test LIBCXX_SUPPORTS_ZL_FLAG - Failed
-- Performing Test LIBCXX_SUPPORTS_NODEFAULTLIB_FLAG
-- Performing Test LIBCXX_SUPPORTS_NODEFAULTLIB_FLAG - Failed
-- Performing Test LIBCXX_SUPPORTS_STD_EQ_CXX14_FLAG
-- Performing Test LIBCXX_SUPPORTS_STD_EQ_CXX14_FLAG - Success
-- Performing Test HAVE_C_FLAG_-fPIC
-- Performing Test HAVE_C_FLAG_-fPIC - Success
## -----------------------------------------------------------------
## Version: 4.4.2
## Build type: RelWithDebInfo
## Build options: OPT_SIM OPT_Z3 OPT_STP
## -----------------------------------------------------------------
-- Configuring incomplete, errors occurred!
make[2]: *** [Makefile:98: config] Error 1
make[1]: *** [Makefile:120: release-divine] Error 2
make: *** [Makefile:76: release] Error 2
``` 

We might need to look for some more recent and still active model checkers for C++. 

@feliperodri How did we get the LLBMC binary last time?

**Comment by lucasccordeiro** [comment]
@kunjsong01: you could try https://cppcheck.sourceforge.io/.
